### PR TITLE
quiche: update HTTP/3 config creation to new API

### PR DESCRIPTION
**Pull request by LPardue**
Depends on pending changes to quiche: 4a51aad615975a4051ea32e2d25104feac90c430 and 906646c43512efcd651f415b279cb89797c97fb8. See https://github.com/cloudflare/quiche/pull/180/commits for context

### CODE REVIEW

File: lib/vquic/quiche.c [hunk 1]
```diff
@@ -237,10 +237,12 @@ static CURLcode quiche_has_connected(struct connectdata *conn,
   conn->httpversion = 30;
   conn->bundle->multiuse = BUNDLE_MULTIPLEX;
 
-  qs->h3config = quiche_h3_config_new(0, 1024, 0, 0);
+  qs->h3config = quiche_h3_config_new();
   if(!qs->h3config)
     return CURLE_OUT_OF_MEMORY;
 
+  quiche_h3_config_set_max_header_list_size(qs->h3config, 1024);
```
Comment (ghedo) [inline lib/vquic/quiche.c]: I think we can omit this. Hardcoding the limit here is not very useful and AFAICT the ngtcp2 backend doesn't set a limit, so we should probably be consistent.

Comment (LPardue) [reply lib/vquic/quiche.c]: Make sense. For anyone not tracking H3 specs closely, if an endpoint doesn't set a HTTP_MAX_HEADER_LIST_SIZE it is treated as effectively unlimited. Constructing the Config object and not setting anything will invoke that behaviour.

File: lib/vquic/quiche.c [hunk 2]
```diff
@@ -195,7 +195,7 @@ CURLcode Curl_quic_connect(struct connectdata *conn, curl_socket_t sockfd,
   qs->conn = quiche_connect(conn->host.name, (const uint8_t *) qs->scid,
                             sizeof(qs->scid), qs->cfg);
   if(!qs->conn) {
-    failf(data, "can't create quiche connection");
+    failf(data, "can't create quiche connectconfigion");
```
Comment (ghedo) [inline lib/vquic/quiche.c]:
```suggestion
    failf(data, "can't create quiche connection");
```

**Comment by ghedo** [comment]
The quiche PR was merged, so to avoid breaking other curl builds to fail I squashed and pushed this change manually as https://github.com/curl/curl/commit/19338e97216f774976fd6949f4fc675475f528b7 even if the builds are not finished yet (I tested this locally first). If something breaks I'll look into preparing a fix.

**Comment by ghedo** [comment]
Aaaaand I forgot to add "Closes: ..." to the commit message :man_facepalming:

**Pull request closed without merging**